core/state_processor_test.go

```diff
@@ -518,11 +519,13 @@ func TestProcessVerkle(t *testing.T) {
 	//f.Write(buf.Bytes())
 	//fmt.Printf("root= %x\n", chain[0].Root())
 	// check the proof for the last block
-	err := trie.DeserializeAndVerifyVerkleProof(proofs[1], chain[0].Root().Bytes(), chain[1].Root().Bytes(), keyvals[1])
-	if err != nil {
-		t.Fatal(err)
+
+	for i := 1; i < len(proofs); i++ {
+		if err := trie.DeserializeAndVerifyVerkleProof(proofs[i], statediff[i], chain[i-1].Root().Bytes(), computedKeys[i], computedPreStateValues[i], computedPostStateValues[i]); err != nil {
+			t.Fatal(err)
+		}
+		t.Log("verfied verkle proof")
```

Comment:
Now `DeserializeAndVerifyVerkleProof` expects the caller to provide trusted values of:
- Keys
- Pre-state
- Post-state

It will use that to verify the block proof `StateDiff` is correct.

Also, generalized this proof verification check for all the generated blocks in the test (and not only the block 1). If we extend this tests with more blocks, all should pass.